x/poolmanager/types/hooks.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import sdk "github.com/cosmos/cosmos-sdk/types"
+
+type PoolManagerHooks interface {
+	// AfterPoolCreated is called after CreatePool
+	AfterPoolCreated(ctx sdk.Context, sender sdk.AccAddress, poolId uint64)
+}
+
+var _ PoolManagerHooks = MultiPoolManagerhooks{}
+
+// combine multiple PoolManager hooks, all hook functions are run in array sequence.
+type MultiPoolManagerhooks []PoolManagerHooks
```

Comment:
What's the reason for having multiple combined hooks here?

Reply:
it's so that we can have multiple hook recievers for each hook. For example: Gamm has 2 hook recievers poolIncentive and twap. At the end of a specific function in Gamm we can trigger something in both poolIncentives & twap